sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -82,18 +82,19 @@ private[sql] object Dataset {
     dataset
   }
 
-  def ofRows(sparkSession: SparkSession, logicalPlan: LogicalPlan): DataFrame = {
-    val qe = sparkSession.sessionState.executePlan(logicalPlan)
-    qe.assertAnalyzed()
-    new Dataset[Row](sparkSession, qe, RowEncoder(qe.analyzed.schema))
+  def ofRows(sparkSession: SparkSession, logicalPlan: LogicalPlan): DataFrame =
+    sparkSession.withActive {
```

Comment:
do we need to scope the active session here? The operations below, `sparkSession.sessionState.executePlan` and `new Dataset`, both seem to scope the active session themselves.

Reply:
I see, the creation of `RowEncoder` may access active session

Reply:
shall we only wrap it?

Reply:
nvm, it's too tricky to find out which part uses active session. It's safer to wrap the entire method.